### ISSUE TITLE
Double clicking a session file on the mac starts a blank session

### ISSUE THREAD

**Issue by AdamCarroll** [Santulator/Santulator#9]
Once the Santulator installer has been run, double clicking on a `.santa` session file opens Santulator.  This works on all three supported platforms.  Unfortunately, on the Mac, Santulator starts with a blank session instead of the contents of the session file.

Note that this problem does not exist on Windows or Linux.

**Comment by xzel23** [comment]
There seems to be a solution to this in the accepted answer to [this](https://stackoverflow.com/questions/29101472/pass-parameters-to-javafx-application-by-double-click-on-file) stackoverflow question.

**Comment by AdamCarroll** [comment]
Thanks @xzel23 .  That's interesting.  To be honest, I thought the `com.apple.eawt` classes had gone away in JDK 11 but just checking now, it looks like I have them there in my local Java 11 from AdoptOpenJDK.  It would be interesting to see if this could be used as a solution and indeed whether those classes are here to stay.  Have you tried this out in any of your own projects?

**Comment by xzel23** [comment]
Hi @AdamCarroll . No, I did not find the time yet. But I will try next month.

**Comment by xzel23** [comment]
H @AdamCarroll , I finally have reached the point where this gets really important, so I tried for several days to find a usable workaround, but to no avail. So I forked JavaFX and four hours later I have a working version (some changes to the `com.sun.glass.ui.Application` and `com.sun.javafx.tk.quantum.QuantumToolkit`classes plus adding a `setOpenFilesHandler()` method to `javafx.application.Application`). I just sent a mail to the OpenJFX dev list asking for comments and I hope it will find its way in. It is really unbelievable something as important as opening applications by double clicking on the associated file types did not work for such a long time on Mac.

In short, you don't use Desktop.setOpenFilesHandler() methods, but call setOpenFilesHandler from your application class. (Just calling Desktop.getDesktop() breaks the global menu on Mac, so I think it's good not to have to call it).

**Comment by AdamCarroll** [comment]
That's very interesting.  Let's see what response you get.  Do you have a link you can share with your report?

**Comment by xzel23** [comment]
Sorry, I am really busy after taking a contract now. There's a patched JavaFX 14 at my GitHub that you can try: https://github.com/xzel23/jfx/tree/jfx14.

You have to register the file open handler with the new method in the Application class of JavaFX and *not* use awt.Desktop. Should work as expected. I had written to the OpenJFX mailing list suggesting this as a patch, but there was no reaction, and I didn't find the time to go on with this.
